src/content/docs/highlights/v0.30.md

```diff
@@ -0,0 +1,692 @@
+text += Text::from("line 2");
+```
+
+Style and alignment applied to the second text is ignored (though styles and alignment of lines and
+spans are copied).
+
+- Implement `UnicodeWidthStr` for `Text`/`Line`/`Span` for retrieving the width via `width` and
+  `width_cjk`
+
+---
+
+## Styling 🎨
+
+- Support conversions from [anstyle](https://crates.io/crates/anstyle) styles (gated behind
+  `anstyle` feature)
+
+```rust
+let anstyle_color = anstyle::Ansi256Color(42);
+let color = Color::from(anstyle_color);
+```
+
+- Add generic color conversion methods from tuples
+
+```rust
+Color::from([255, 0, 0]);
+Color::from((255, 0, 0));
+Color::from([255, 0, 0, 255]);
+Color::from((255, 0, 0, 255));
+```
+
+- Implement `Styled` for primitives such as `u8`, `i32`, `f64`, `Cow<'a, str>`, etc.
+
+```rust
+let s = Cow::Borrowed("a");
+assert_eq!(s.red(), "a".red());
+```
+
+- Implement stylize methods directly on `Style`
+
+This makes it possible to create constants using the shorthand methods.
+
+```rust
+const MY_STYLE: Style = Style::new().blue().on_black();
+```
+
+This is a
+[breaking change](https://github.com/ratatui/ratatui/blob/main/BREAKING-CHANGES.md#style-no-longer-implements-styled-1572).
+
+---
+
+## Layout 📐
+
+- Add ergonomic methods to `Rect` for layouting
+
+TODO: Make this section read better
+<https://github.com/ratatui/ratatui/commit/6dcd53bc6ba172d4b96d02809c6a46105e67d85a>
+
+We introduced new methods for `Rect` that simplify the process of splitting a `Rect` into sub-rects
+according to a given `Layout`.
+
+`Rect::layout` and `Rect::try_layout`:
+
+```rust
+use ratatui_core::layout::{Layout, Constraint, Rect};
+let area = Rect::new(0, 0, 10, 10);
+let layout = Layout::vertical([Constraint::Fill(1); 2]);
+
+// Rect::layout() infers the number of constraints at compile time:
+let [top, main] = area.layout(&layout);
+```
+
+`Layout::try_areas` method that returns an array of sub-rects, with compile-time checks for the
+number of constraints.
+
+This is added mainly for consistency with the new `Rect` methods.
+
+```rust
+// Rect::try_layout() and Layout::try_areas() do the same, but return a
+// Result:
+let [top, main] = area.try_layout(&layout)?;
+let [top, main] = layout.try_areas(area)?;
+```
+
+`Rect::layout_vec` method that returns a `Vec` of sub-rects.
+
+```rust
+let areas_vec = area.layout_vec(&layout);
+```
+
+- Add helper methods to `Rect` for centering
+
+For centering:
+
+```rust
+let area = frame
+  .area()
+  .centered(Constraint::Ratio(1, 2), Constraint::Ratio(1, 3));
+```
+
+Or for vertical centering:
+
+```rust
+let area = frame.area().centered_vertically(Constraint::Ratio(1, 2));
+```
+
+Horizontally:
+
+```rust
+let area = frame.area().centered_horizontally(Constraint::Length(3));
+```
+
+- Add `Rect::outer` method
+
+This creates a new `Rect` outside the current one, with the given margin applied on each side.
+
+TODO: fix typo in https://github.com/ratatui/ratatui/pull/1929 and add visual tests & examples
```

Comment:
What's the typo?

Reply:
https://github.com/ratatui/ratatui/pull/2123